ui-backend/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/cql/SourceWarningsFilterManager.java

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.query.cql;
+
+import ddf.catalog.operation.ProcessingDetails;
+import ddf.catalog.operation.impl.ProcessingDetailsImpl;
+import ddf.catalog.security.SourceWarningsFilter;
+import java.util.List;
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+/**
+ * A {@code SourceWarningsFilterManager} {@linkplain #filterWarningsOf(ProcessingDetails) filters
+ * the {@code warnings} of} {@link ProcessingDetails} with the {@linkplain
+ * #getBestCompatibleFilterFor(ProcessingDetails) best compatible} {@link SourceWarningsFilter} in
+ * its {@code filters}} such that the {@link ProcessingDetails} only contain {@code warnings} which
+ * the {@link SourceWarningsFilter} deems useful to the user.
+ */
+public class SourceWarningsFilterManager {
+
+  private List<SourceWarningsFilter> filters;
+
+  public SourceWarningsFilterManager(List<SourceWarningsFilter> filters) {
+    if (filters == null) {
+      throw new IllegalArgumentException(
+          "the constructor of SourceWarningsFilterManager received a null argument");
+    }
+
+    this.filters = filters;
+  }
+
+  public ProcessingDetails filterWarningsOf(ProcessingDetails processingDetails) {
+    SourceWarningsFilter bestCompatibleFilter = null;
+
+    if (canInspect(processingDetails)) {
+      bestCompatibleFilter = getBestCompatibleFilterFor(processingDetails);
+    }
+
+    return bestCompatibleFilter == null
+        ? new ProcessingDetailsImpl()
+        : bestCompatibleFilter.filter(processingDetails);
```

Comment:
❗ Instead of maintaining intermediate state, `return` early. It'll cut down on the code. It also helps keep branching complexity low. 
```
if (canInspect(processingDetails)) {
  return getBestCompatibleFilterFor(processingDetails).filter(processingDetails);
}
return new ProcessingDetailsImpl();
```
This was my first instinct. But, this doesn't work because `getBestCompatibleFilterFor` is  `@Nullable`. What do you think of this instead? Notice the negated predicate. 
```
if (!canInspect(processingDetails)) {
  return new ProcessingDetailsImpl();
}
return Optional.ofNullable(getBestCompatibleFilterFor(processingDetails))
  .map(f -> f.filter(processingDetails))
  .orElse(new ProcessingDetailsImpl());
```